SU-HSDO/suhumsci
# HSD8-576 view field title

# READY FOR REVIEW

# Summary
- Override the viewfield field type, widget and formatter to add the ability for the user to display and customize the title of the view if they so choose to.

# Needed By (Date)
- 2/13

# Urgency
- Medium

# Steps to Test
1. Checkout this branch
1. `composer install --prefer-source`
1. `blt drupal:sync --site=[sitename]`
1. add a new basic page and add a view field component
1. ensure the checkbox "Display view title", check that
1. ensure another checkbox appears to "Override Title", check that
1. ensure a text field appears for the "Overridden Title", populate that as desired
1. save and ensure your custom title displays above the view. if the view has no result, ensure no title is displayed.


# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
